src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -0,0 +1,518 @@
+                fields = default;
+                return false;
+            }
+
+            const int PostebStackBufferSize = 256;
+            Span<char> postebStackBuffer = stackalloc char[PostebStackBufferSize];
+            int areaOffset = 0;
+            int preebIndex;
+            while ((preebIndex = pemData.IndexOfByOffset(PreEBPrefix, areaOffset)) >= 0)
+            {
+                int labelStartIndex = preebIndex + PreEBPrefix.Length;
+
+                // If there are any previous characters, the one prior to the PreEB
+                // must be a white space character.
+                if (preebIndex > 0 && !char.IsWhiteSpace(pemData[preebIndex - 1]))
+                {
+                    areaOffset += labelStartIndex;
+                    continue;
+                }
+
+                int preebEndIndex = pemData.IndexOfByOffset(Ending, labelStartIndex);
+
+                // There is no ending sequence, -----, in the remainder of
+                // the document. Therefore, there can never be a complete PreEB
+                // and we can exit.
+                if (preebEndIndex < 0)
+                {
+                    fields = default;
+                    return false;
+                }
+
+                Range labelRange = labelStartIndex..preebEndIndex;
+                ReadOnlySpan<char> label = pemData[labelRange];
+
+                // There could be a preeb that is valid after this one if it has an invalid
+                // label, so move from there.
+                if (!IsValidLabel(label))
+                {
+                    goto NextAfterLabel;
+                }
+
+                int contentStartIndex = preebEndIndex + Ending.Length;
+                int postebLength = PostEBPrefix.Length + label.Length + Ending.Length;
+
+                Span<char> postebBuffer = postebLength > PostebStackBufferSize
+                    ? new char[postebLength]
+                    : postebStackBuffer;
+                ReadOnlySpan<char> posteb = WritePostEB(label, postebBuffer);
+                int postebStartIndex = pemData.IndexOfByOffset(posteb, contentStartIndex);
+
+                if (postebStartIndex < 0)
+                {
+                    goto NextAfterLabel;
+                }
+
+                int pemEndIndex = postebStartIndex + postebLength;
+
+                // The PostEB must either end at the end of the string, or
+                // have at least one white space character after it.
+                if (pemEndIndex < pemData.Length - 1 &&
+                    !char.IsWhiteSpace(pemData[pemEndIndex]))
+                {
+                    goto NextAfterLabel;
+                }
+
+                Range contentRange = contentStartIndex..postebStartIndex;
+
+                if (!TryCountBase64(pemData[contentRange],
+                                   out int base64start,
+                                   out int base64end,
+                                   out int decodedSize))
```

Comment:
These look like they'd all fit on one line for the normal github aperture.  If you want/need the newlines, chop before the first argument and only indent 4 spaces.